checkmarx/util.py

```diff
@@ -176,6 +179,15 @@ def init_scan(
             finished += 1
             logger.info(f'remaining: {artifact_count - finished}')
 
+    try:
+        upload_results_to_deliverydb(
```

Comment:
please mv this to the step-implementation. doing such operations implicitly I consider a bad pattern (admittedly one that we also have in other places)

consider the name of the method you added this code to - it reads `init_scan`. this is strongly misleading.

instead, please rework the code like so (pseudo-pseudo-code):

```
scan_results = scan_with_checkmarx(source_nodes_to_scan)
update_github_issues(scan_results)
upload_to_delivery_service(scan_results)
```